### PR TITLE
Refactor C++ unit tests to allow finer grained filtering

### PR DESCRIPTION
**Pull request by ChuckHastings**
Running all of the C++ unit tests can take a long time.

This PR splits the file-based tests into small versus large tests.  The large tests (usually named file_large_test) can be skipped by setting the following environment variable:

`export GTEST_FILTER="-*large*"`

You can skip both the large tests and the benchmark tests with the following:

`export GTEST_FILTER="-*benchmark*:*large*"`

By default all tests will be run.

### CODE REVIEW

**Review by jnke2016 — APPROVED** [review]

File: cpp/tests/centrality/eigenvector_centrality_test.cpp [hunk 1]
```diff
@@ -277,8 +277,18 @@ INSTANTIATE_TEST_SUITE_P(
                       EigenvectorCentrality_Usecase{500, false, true},
                       EigenvectorCentrality_Usecase{500, true, false},
                       EigenvectorCentrality_Usecase{500, true, true}),
-    ::testing::Values(cugraph::test::File_Usecase("test/datasets/karate.mtx"),
-                      cugraph::test::File_Usecase("test/datasets/web-Google.mtx"),
+    ::testing::Values(cugraph::test::File_Usecase("test/datasets/karate.mtx"))));
+
+INSTANTIATE_TEST_SUITE_P(
+  file_large_pass,
```
Comment (seunghwak) [inline cpp/tests/centrality/eigenvector_centrality_test.cpp:283]: Why are we using "file_large_test" above and "file_large_pass" here? Better be consistent?

**Review by seunghwak — APPROVED** [review]
Besides the minor comment about the inconsistency, LGTM.

File: cpp/tests/centrality/eigenvector_centrality_test.cpp (hunk 1 above)
Comment (ChuckHastings) [reply cpp/tests/centrality/eigenvector_centrality_test.cpp]: Updated eigenvector centrality test names to match.

**Pull request merged**